components/server/src/bitbucket-server/bitbucket-server-repository-provider.ts

```diff
@@ -146,7 +146,9 @@ export class BitbucketServerRepositoryProvider implements RepositoryProvider {
 
     async getUserRepos(user: User): Promise<string[]> {
         try {
-            const repos = await this.api.getRepos(user, { permission: "REPO_READ" });
+            // TODO: See if there's another api we could use here, such as recent repos for the user
+            // Only grab one page of up to 100 repos
+            const repos = await this.api.getRepos(user, { limit: 100, cap: 1, permission: "REPO_READ" });
```

Comment:
I saw you change this file too, it seems dashboard will only query once, so if we add limit 100 here, in our case, we can't find repos starts with `ten`

|IMG|
|:-:|
|<img width="374" alt="image" src="https://github.com/gitpod-io/gitpod/assets/20944364/5351b2a0-6d47-47eb-b4fe-55984e7787b4">|

Reply:
Not a big blocker since we can force input repo url here.

Reply:
If you're looking for recent repos, this API should be a good way to do it https://developer.atlassian.com/server/bitbucket/rest/v811/api-group-repository/#api-api-latest-profile-recent-repos-get

See demo https://bitbucket.gitpod-dev.com/rest/api/1.0/profile/recent/repos

Reply:
Discussed this with @akosyakov - and currently that's similar to what we do with GitHub, where we fetch ~ 100 repos (higher chance the GitHub ones match what you want due to a more robust api). This aligns it closer to that, but it does mean there's a chance you won't find your repo. The workarounds are to either paste the url from in, or create a project, and then it would match.

Reply:
Sorry submitted that before your other comments showed up - but yah, recent repos seems like it could be better here, but lets do that in a followup.

Reply:
improvement issue create https://linear.app/gitpod/issue/EXP-513